torch/testing/_internal/common_utils.py

```diff
@@ -777,27 +777,65 @@ def safeCoalesce(self, t):
 
         return tg
 
-    def assertEqual(self, x, y, prec=None, message='', allow_inf=False, exact_dtype=None):
+    # Some analysis of tolerance by logging tests from test_torch.py can be found
+    # in https://github.com/pytorch/pytorch/pull/32538.
+    dtype_precisions = {
+        torch.half: 0.001,
+        torch.bfloat16: 0.016,
+        torch.float: 1.3e-6,
+        torch.double: 5e-7,
+        torch.int: 0
+    }
+
+    def compared_dtype(self, x, y):
+        dtype = torch.float
+        if isinstance(x, torch.Tensor):
+            dtype = x.dtype
+        elif isinstance(y, torch.Tensor):
+            dtype = y.dtype
```

Comment:
This suggests some unpleasant directionality in assertEqual, where assertEqual(x, y) is not the same thing as assertEqual(y, x). It's true that this directionality exists today (where we run y.to(x)), but I'm wondering if we shouldn't be doing something like, setting `compared_dtype` to the lowest precision out of all the dtypes involved.

Reply:
yup, I'm thinking to make this more like _get_default_tolerance from testing/__init__.py (and then getting rid of that one in a future PR).